File: app/Models/Tenant/Event.php
<?php

namespace App\Models\Tenant;

use Illuminate\Database\Eloquent\Builder;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\SoftDeletes;

class Event extends Model
{
    use SoftDeletes;

    protected $connection = 'society';

    public const CATEGORIES = ['cultural', 'sports', 'meeting', 'festival', 'other'];

    protected $fillable = [
        'title',
        'description',
        'category',
        'location',
        'start_at',
        'end_at',
        'status',
        'posted_by_user_id',
    ];

    protected $casts = [
        'start_at' => 'datetime',
        'end_at' => 'datetime',
    ];

    public function postedBy(): BelongsTo
    {
        return $this->belongsTo(User::class, 'posted_by_user_id');
    }

    /**
     * Scopes
     */
    public function scopeUpcoming(Builder $query): Builder
    {
        return $query->where('status', 'published')->where('start_at', '>=', now());
    }

    public function scopePast(Builder $query): Builder
    {
        return $query->where('status', 'published')->where('start_at', '<', now());
    }

    public function scopeCancelled(Builder $query): Builder
    {
        return $query->where('status', 'cancelled');
    }

    public function isPast(): bool
    {
        return $this->start_at->isPast();
    }

    public function isCancelled(): bool
    {
        return $this->status === 'cancelled';
    }
}

File: mobile/lib/features/events/data/event.dart
class Event {
  Event({
    required this.id,
    required this.title,
    this.description,
    required this.category,
    this.location,
    this.startAt,
    this.endAt,
    required this.status,
    this.postedBy,
  });

  factory Event.fromJson(Map<String, dynamic> json) {
    return Event(
      id: json['id'] as int,
      title: json['title'] as String,
      description: json['description'] as String?,
      category: json['category'] as String,
      location: json['location'] as String?,
      startAt: json['start_at'] as String?,
      endAt: json['end_at'] as String?,
      status: json['status'] as String,
      postedBy: json['posted_by'] as String?,
    );
  }

  final int id;
  final String title;
  final String? description;
  final String category;
  final String? location;
  final String? startAt;
  final String? endAt;
  final String status;
  final String? postedBy;
}
